### PR TITLE
[openvdb] update to v8.0.0

### LINKED ISSUE #15811: [openvdb] update to v8.0.0
Library name: OpenVDB

New version number: v8.0.0

Other information that may be useful (release notes, etc...)

* Release notes: https://github.com/AcademySoftwareFoundation/openvdb/releases/tag/v8.0.0
* v8.0.0 comes with a new tool/extension called [OpenVDB AX](https://www.openvdb.org/documentation/doxygen/openvdbax.html). This could be added as a feature package.

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- What does your PR fix? Fixes #15811 

Update openvdb to the latest version 8.0.0 and add AX feature

All features are tested successfully in the following trip;ets:
- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

**Comment by orbingol** [comment]
I was wondering if there are any updates on this PR.

I also would like to add a note: OpenVDB compiles and links against the latest version of blosc on Windows as well as v1.5.0 of blosc library. However, it requires blosc v1.5.0 on Linux (tested with Ubuntu 18.04) and macOS (tested with Catalina). It compiles but does not link when blosc is NOT v1.5.0 on Linux and macOS.

Is there a way to add blosc v1.5.0 instead of the latest available via vcpkg?

**Comment by JonLiu1993** [comment]
> I was wondering if there are any updates on this PR.
> 
> I also would like to add a note: OpenVDB compiles and links against the latest version of blosc on Windows as well as v1.5.0 of blosc library. However, it requires blosc v1.5.0 on Linux (tested with Ubuntu 18.04) and macOS (tested with Catalina). It compiles but does not link when blosc is NOT v1.5.0 on Linux and macOS.
> 
> Is there a way to add blosc v1.5.0 instead of the latest available via vcpkg?

@orbingol ,If you need to use blosc v1.5.0 version, you can submit an update issue, I can update to the latest version and fix this pr together.

**Comment by orbingol** [comment]
@JonLiu1993 I mean OpenVDB needs to use [blosc v1.5.0](https://github.com/Blosc/c-blosc/releases/tag/v1.5.0) for Linux and macOS. I believe blosc v1.5.0 would be a downgrade for the existing port, v1.5.0 was released on Nov 7th, 2014 according to the GH release page.

Regarding OpenVDB, here are some references:

* Linux install guide instructing to use blosc v1.5.0: https://github.com/AcademySoftwareFoundation/openvdb#linux
* OpenVDB CMakeLists.txt about blosc: https://github.com/AcademySoftwareFoundation/openvdb/blob/86e07639320848e25cef3e663a35f0b68680b32b/CMakeLists.txt#L203

Since OpenVDB needs this specific version of blosc, I was wondering if it is possible to create a port, like `blosc150`, for use with OpenVDB till they support a more recent version of blosc. blosc v1.5.0 could also be integrated into the build step of OpenVDB. I don't think any other library would depend on such an old version of blosc. I can surely open an issue referencing this PR.

I am guessing installing different versions of the ports are still WIP in vcpkg, and creating a port as I suggested might solve OpenVDB Linux and macOS issues till the official solution comes. It looks like you were also able to compile against the windows triples.

**Review by ytbigvinny27 — APPROVED** [review]

File: ports/openvdb/portfile.cmake [hunk 1]
```diff
@@ -25,6 +25,14 @@ if ("tools" IN_LIST FEATURES)
   endif()
 endif()
 
+if ("ax" IN_LIST FEATURES)
+  if(!WIN32)
```
Comment (NancyLi1013) [inline ports/openvdb/portfile.cmake:29]:
```suggestion
  if(NOT VCPKG_TARGET_IS_WINDOWS)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
>@JonLiu1993 I mean OpenVDB needs to use blosc v1.5.0 for Linux and macOS. I believe blosc v1.5.0 would be a downgrade for the existing port, v1.5.0 was released on Nov 7th, 2014 according to the GH release page.

You can add another version; accessing it though would require a specific overridden version.

> I am guessing installing different versions of the ports is still WIP in vcpkg

Different versions of ports at the same time is fundamentally impossible; it isn't a vcpkg limitation, it is a C++ limitation. It breaks the ODR.

That all seems like a preexisting problem not directly related to this PR so I'll merge it once build passes on the current infrastructure.

**Comment by BillyONeal** [comment]
Thanks for your contribution! :)

**Pull request merged**